testtools/genblocks/main.go

```diff
@@ -101,16 +107,18 @@ func main() {
 		}
 
 		headerBytes, err := blockHeader.MarshalBinary()
+		if err != nil {
+			panic(fmt.Sprint("Cannot marshal block header: ", err))
+		}
+		fmt.Print(hex.EncodeToString(headerBytes))
 
 		// After the header, there's a compactsize representation of the number of transactions.
-		if numTransactions >= 253 {
-			panic("Sorry, this tool doesn't support more than 253 transactions per block.")
+		if numTransactions < 253 {
+			fmt.Printf("%02x", numTransactions)
+		} else {
+			fmt.Printf("%02x%02x%02x", 253, numTransactions%256, numTransactions/256)
```

Comment:
Using `%02x` seems a little simpler than `hex.EncodeToString()`, but the effect is the same. It's important to get the endianness correct here, the least-significant digit comes first (little-endian).